src/language_features/document_symbol.rs

```diff
@@ -23,6 +23,125 @@
+                Some((server_name, Some(DocumentSymbolResponse::Nested(symbols))))
+                    if !symbols.is_empty() =>
+                {
+                    let server = &ctx.language_servers[&server_name];
+                    let document = if let Some(document) = ctx.documents.get(&meta.buffile) {
+                        document
+                    } else {
+                        return;
+                    };
+                    let mut filename_path = PathBuf::default();
+                    let symbol = &symbols[0];
+                    let filename = symbol_filename(&meta, symbol, &mut filename_path).to_string();
+                    let mut breadcrumbs = Vec::default();
+                    breadcrumbs_calc(
+                        &symbols,
+                        &params,
+                        ctx,
+                        server,
+                        document,
+                        filename.as_str(),
+                        &mut breadcrumbs,
+                    );
+                    if let Some(last) = breadcrumbs.last_mut() {
+                        *last = format!("{last} ");
+                    };
+                    let mut breadcrumbs = breadcrumbs.join(" > ");
+                    breadcrumbs = editor_quote(breadcrumbs.as_str());
+                    let command =
+                        format!("set-option buffer lsp_modeline_breadcrumbs {breadcrumbs}");
+                    let command = format!(
+                        "evaluate-commands -buffer {} -- {}",
+                        editor_quote(&meta.buffile),
+                        editor_quote(&command)
+                    );
+                    ctx.exec(meta, command);
+                }
+                Some((_, Some(DocumentSymbolResponse::Nested(_)))) => {
+                    return;
+                }
+                Some((_, Some(DocumentSymbolResponse::Flat(_)))) => {
+                    return;
+                }
+                Some((_, None)) => {
+                    return;
+                }
+                None => {
+                    return;
+                }
+            };
+        },
+    );
+}
+
+fn breadcrumbs_calc(
+    symbols: &[DocumentSymbol],
+    params: &BreadcrumbsParams,
+    ctx: &Context,
+    server: &ServerSettings,
+    document: &Document,
+    filename: &str,
+    acc: &mut Vec<String>,
+) {
+    for symbol in symbols {
+        let symbol_range = get_kakoune_range(server, filename, &symbol.range, ctx).unwrap();
+        let is_inside = {
+            let is_after_symbol_start = symbol_range.start.line <= params.position_line;
+            let is_before_symbol_end = params.position_line <= symbol_range.end.line;
+            is_after_symbol_start && is_before_symbol_end
+        };
+        if is_inside {
+            match symbol.kind() {
+                SymbolKind::FUNCTION | SymbolKind::METHOD => {
+                    acc.push(symbol.name.clone() + "()");
```

Comment:
not sure if the `()` make sense everywhere -- it's not really kakoune-lsp's job to be creative about this kind of thing -- but I guess we can leave it and improve this later

Reply:
Yes, in some languages it's possible to define parentheses-less methods which doesn't look consistent.

Parentheses are nice for languages where function or method indeed always defined with `foo()`.

I'm voting for removing this logic and to make consistent across all languages.

If we look at VS Code they assign a specific emoji/symbol to all `SymbolKind`s. Honestly, I don't see a connection, e.g. between a cube and `method`, but I guess it's a repetition and consistency which helps to quickly see where is what.